samples/Mvc.FileUpload/Controllers/FileController.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
+using Microsoft.WindowsAzure.Storage;
+using Mvc.FileUpload.Filters;
+using Mvc.FileUpload.Models;
+
+namespace Mvc.FileUpload.Controllers
+{
+    public class FileController : Controller
+    {
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly ILogger<FileController> _logger;
+
+        // Get the default form options so that we can use them to set the default limits for
+        // request body data
+        private static readonly FormOptions _defaultFormOptions = new FormOptions();
```

Comment:
In this sample request form is being parsed manually (in the sense that MultipartReader is being used to read individual parts and handled) compared to depending on FormFeature. Most of the code here is borrowed from FormFeature and I wanted to use the default limits here too.